Model/Kount/EnsConfig.php

```diff
@@ -0,0 +1,361 @@
+
+                        if ((string) $event->new_value === self::RESPONSE_DECLINE) {
+                            return $this->declineOrder($order);
+                        }
+                    }
+                }
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param $event
+     * @return int|null
+     */
+    public function getIncrementId($event)
+    {
+        if (isset($event->key['order_number'])) {
+            return (int) $event->key['order_number'];
+        }
+
+        return null;
+    }
+
+    /**
+     * @param $event
+     * @return string|null
+     */
+    public function getKountTransactionId($event)
+    {
+        if (isset($event->key)) {
+            return (string) $event->key;
+        }
+
+        return null;
+    }
+
+    /**
+     * @param OrderInterface $order
+     * @return bool
+     * @throws Exception
+     */
+    public function approveOrder(OrderInterface $order): bool
+    {
+        /** @var Order $order */
+        if ($order->getStatus() === Order::STATUS_FRAUD || $order->getStatus() === Order::STATE_PAYMENT_REVIEW) {
+            $order->getPayment()->accept();
+            $this->orderRepository->save($order);
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param OrderInterface $order
+     * @return bool
+     * @throws Exception
+     */
+    public function declineOrder(OrderInterface $order): bool
+    {
+        /** @var Order $order */
+        if ($order->getStatus() === Order::STATUS_FRAUD || $order->getStatus() === Order::STATE_PAYMENT_REVIEW) {
+            $braintreeId = $order->getPayment()->getCcTransId();
+
+            /** @var Transaction $braintreeTransaction */
+            $braintreeTransaction = $this->braintreeAdapter->findById($braintreeId);
+
+            if ($braintreeTransaction) {
+                if ($braintreeTransaction->status === Transaction::AUTHORIZED
+                    || $braintreeTransaction->status === Transaction::SUBMITTED_FOR_SETTLEMENT) {
+                    return $this->voidOrder($order);
+                }
+
+                if ($braintreeTransaction->status === Transaction::SETTLED) {
+                    return $this->refundOrder($order);
+                }
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param OrderInterface $order
+     * @return bool
+     * @throws Exception
+     */
+    public function voidOrder(OrderInterface $order): bool
+    {
+        /** @var Collection $invoices */
+        $invoices = $order->getInvoiceCollection();
+
+        if (count($invoices->getItems()) > 0) {
+            foreach ($invoices as $invoice) {
+                /** @var Invoice $invoice */
+                $invoice->void();
+                $invoice->getOrder()->setStatus(Order::STATE_CANCELED);
+                $invoice->getOrder()->addCommentToStatusHistory(
+                    'Order declined through Kount, order voided in Magento.'
```

Comment:
Unsure if this needs to be translatable?

Reply:
It can't hurt to make it translatable? I'll add it in 👍